yt_dlp/extractor/youtube.py

```diff
@@ -627,6 +627,16 @@ def _extract_account_syncid(*args):
                 # and just "user_syncid||" for primary channel. We only want the channel_syncid
                 return sync_ids[0]
 
+    @staticmethod
+    def _extract_visitor_data(*args):
+        """
+        Extracts visitorData from an API response or ytcfg
+        Appears to be used to track session state
+        """
+        return traverse_obj(
+            args, (..., ('VISITOR_DATA', ('INNERTUBE_CONTEXT', 'client', 'visitorData'), ('responseContext', 'visitorData'))),
```

Comment:
are you sure this works? I remember `traverse_obj`' having only 3 levels of nesting

Reply:
I tested this in a separate script and it appeared to